### PR TITLE
Unmount the share before deleting it, so we don't try to use it later on

### LINKED ISSUE #23686: files:transfer-ownership command does not work in some scenarios with shares
### Steps to reproduce

1. Create 3 oC users, user1, user2 and user3
2. Log in with user1
3. Create a file.txt
4. Share the file with user2 and user3
5. As admin via CLI run  -->  sudo -u www-data ./occ files:transfer-ownership user1 user2

### Expected behaviour

The ownership of files should be transferred from user1 to user2 

### Actual behaviour

An error is shown and the server is no longer available for user2 and user3

### Server configuration
**Operating system**:
Ubuntu 14.04

**Web server:**
Apache

**Database:**
MySQL

**PHP version:**
5.5.9

**ownCloud version:** "9.0.1.1","versionstring":"9.0.1 RC1","edition":"Enterprise"}

**Updated from an older ownCloud or fresh install:**
Fresh

**List of activated apps:**
Enabled:
  - activity: 2.2.1
  - admin_audit: 0.7
  - comments: 0.2
  - dav: 0.1.6
  - enterprise_key: 0.1.0
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_external: 0.5.2
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firewall: 2.3.0
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0
  - windows_network_drive: 0.2.33
  - workflow: 0.1.3
Disabled:
  - announcementcenter
  - encryption
  - external
  - files_antivirus
  - files_drop
  - files_ldap_home
  - objectstore
  - password_policy
  - sharepoint
  - user_external
  - user_ldap
  - user_shibboleth

**The content of config/config.php:**

```

```

**Are you using external storage, if yes which one:** local/smb/sftp/...
No

**Are you using encryption:** 
No

**Logs**
```{"reqId":"UdDGb2IuzHRBFC16CDPi","remoteAddr":"82.159.139.58","app":"PHP","message":"Class 'OC_Mount_Config' not found at \/opt\/owncloud\/lib\/private\/server.php#1239","level":3,"time":"2016-03-31T08:09:38+00:00","method":"GET","url":"\/","user":"--"}
{"reqId":"g\/hlx\/ztIOkoGpA6Ztp7","remoteAddr":"","app":"no app in context","message":"Exception: {\"Exception\":\"OCP\\\\Files\\\\NotFoundException\",\"Message\":\"File with id \\\"21\\\" has not been found.\",\"Code\":0,\"Trace\":\"#0 \\\/opt\\\/owncloud\\\/apps\\\/files_sharing\\\/lib\\\/sharedstorage.php(90): OC\\\\Files\\\\View->getPath(21)\\n#1 \\\/opt\\\/owncloud\\\/apps\\\/files_sharing\\\/lib\\\/sharedstorage.php(581): OC\\\\Files\\\\Storage\\\\Shared->init()\\n#2 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/wrapper.php(381): OC\\\\Files\\\\Storage\\\\Shared->getCache('', Object(OCA\\\\Workflow\\\\AutoTagging\\\\StorageWrapper))\\n#3 \\\/opt\\\/owncloud\\\/apps\\\/workflow\\\/autotagging\\\/storagewrapper.php(389): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->getCache('', Object(OCA\\\\Workflow\\\\AutoTagging\\\\StorageWrapper))\\n#4 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/files\\\/view.php(1302): OCA\\\\Workflow\\\\AutoTagging\\\\StorageWrapper->getCache('')\\n#5 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/files\\\/node\\\/root.php(179): OC\\\\Files\\\\View->getFileInfo('\\\/user2')\\n#6 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/files\\\/node\\\/root.php(341): OC\\\\Files\\\\Node\\\\Root->get('\\\/user2')\\n#7 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/share20\\\/share.php(127): OC\\\\Files\\\\Node\\\\Root->getUserFolder('user2')\\n#8 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/share20\\\/manager.php(193): OC\\\\Share20\\\\Share->getNode()\\n#9 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/share20\\\/manager.php(652): OC\\\\Share20\\\\Manager->generalCreateChecks(Object(OC\\\\Share20\\\\Share))\\n#10 \\\/opt\\\/owncloud\\\/apps\\\/files\\\/command\\\/transferownership.php(218): OC\\\\Share20\\\\Manager->updateShare(Object(OC\\\\Share20\\\\Share))\\n#11 \\\/opt\\\/owncloud\\\/apps\\\/files\\\/command\\\/transferownership.php(114): OCA\\\\Files\\\\Command\\\\TransferOwnership->restoreShares(Object(Symfony\\\\Component\\\\Console\\\\Output\\\\ConsoleOutput))\\n#12 \\\/opt\\\/owncloud\\\/3rdparty\\\/symfony\\\/console\\\/Command\\\/Command.php(259): OCA\\\\Files\\\\Command\\\\TransferOwnership->execute(Object(Symfony\\\\Component\\\\Console\\\\Input\\\\ArgvInput), Object(Symfony\\\\Component\\\\Console\\\\Output\\\\ConsoleOutput))\\n#13 \\\/opt\\\/owncloud\\\/3rdparty\\\/symfony\\\/console\\\/Application.php(840): Symfony\\\\Component\\\\Console\\\\Command\\\\Command->run(Object(Symfony\\\\Component\\\\Console\\\\Input\\\\ArgvInput), Object(Symfony\\\\Component\\\\Console\\\\Output\\\\ConsoleOutput))\\n#14 \\\/opt\\\/owncloud\\\/3rdparty\\\/symfony\\\/console\\\/Application.php(192): Symfony\\\\Component\\\\Console\\\\Application->doRunCommand(Object(OCA\\\\Files\\\\Command\\\\TransferOwnership), Object(Symfony\\\\Component\\\\Console\\\\Input\\\\ArgvInput), Object(Symfony\\\\Component\\\\Console\\\\Output\\\\ConsoleOutput))\\n#15 \\\/opt\\\/owncloud\\\/3rdparty\\\/symfony\\\/console\\\/Application.php(123): Symfony\\\\Component\\\\Console\\\\Application->doRun(Object(Symfony\\\\Component\\\\Console\\\\Input\\\\ArgvInput), Object(Symfony\\\\Component\\\\Console\\\\Output\\\\ConsoleOutput))\\n#16 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/console\\\/application.php(145): Symfony\\\\Component\\\\Console\\\\Application->run(NULL, NULL)\\n#17 \\\/opt\\\/owncloud\\\/console.php(88): OC\\\\Console\\\\Application->run()\\n#18 \\\/opt\\\/owncloud\\\/occ(11): require_once('\\\/opt\\\/owncloud\\\/c...')\\n#19 {main}\",\"File\":\"\\\/opt\\\/owncloud\\\/lib\\\/private\\\/files\\\/view.php\",\"Line\":1639}","level":3,"time":"2016-03-31T08:12:31+00:00","method":"--","url":"--","user":"--"}
```
### Client configuration

**browser**

 Firefox

screen shot 2016-03-31 at 10 25 52

screen shot 2016-03-31 at 10 28 13

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #23686 

@DeepDiver1975 @icewind1991 @rullzer @davitol 

@PVince81 I guess could be sev2? and propably 9.0.2 is okay as well.

### CODE REVIEW

**Comment by cmonteroluque** [comment]
@karlitschek need a backport for 9.0.1

**Comment by karlitschek** [comment]
looks good and makes sense. please backport :+1:

File: apps/files/command/transferownership.php [hunk 1]
```diff
@@ -203,9 +203,15 @@ protected function transfer(OutputInterface $output) {
 	private function restoreShares(OutputInterface $output) {
 		$output->writeln("Restoring shares ...");
 		$progress = new ProgressBar($output, count($this->shares));
+		$mountManager = Filesystem::getMountManager($this->destinationUser);
```
Comment (rullzer) [inline apps/files/command/transferownership.php]: Any reason not to inject this?

Comment (nickvergessen) [inline apps/files/command/transferownership.php]: Not really, this was only the quick way i found in the train ;) :bullettrain_front:

File: apps/files/command/transferownership.php [hunk 2]
```diff
@@ -24,7 +24,7 @@
 use OC\Files\Filesystem;
 use OC\Files\View;
 use OCP\Files\FileInfo;
-use OCP\Files\Folder;
```
Comment (nickvergessen) [inline apps/files/command/transferownership.php]: Unused import removed...

**Comment by nickvergessen** [comment]
Setting sev1 as per issue

**Comment by rullzer** [comment]
:+1: good catch!

**Comment by schiesbn** [comment]
tested, works and code looks good :+1:

**Pull request merged**